sdk/cosmos/azure-cosmos/pom.xml

```diff
@@ -132,6 +132,20 @@ Licensed under the MIT License.
       </exclusions>
     </dependency>
 
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-identity</artifactId>
```

Comment:
Is Direct dependency needed?

Reply:
Ohh its a test depdnency?

Reply:
yes, this is a test dependency only; however the second dependency we need to talk about, if we want to take direct dependency on. SimpleTokenCache requires it... basically this creates a background thread via Reactor's ReplayProcessor which ensures that the AAD token is refreshed before it expires.

Reply:
To be more exact, ReplayProcessor requires that an HttpClient implementation is available as part of the current app's package. This can be something that the client bring or our Cosmos library can and the customer can exclude it if they want to bring their own.

Reply:
I am guessing this will be optional dependency right? if AAD integration is needed this would be needed at runtime. no?

shouldn't this be the following?
```xml
    <dependency>
      <groupId>com.azure</groupId>
      <artifactId>azure-identity</artifactId>
      <version>1.1.0-beta.3</version> <!-- {x-version-update;com.azure:azure-identity;dependency} -->
      <optional>true</optional>
    </dependency>
```

Reply:
It's the azure-core-http-netty that might be needed... azure-identity is only required for testing purposes.

Reply:
Regarding the e2e testing, will add these once the AAD support is available in the Cosmos public emulator. Regular AAD requires creating a proper identity first which needs to be recognized by the Cosmos service (this is still worked on); to create this identity the tests need special permission to RBAC (subscription ownership, rights etc). Instead of this we are going to leverage the Cosmos emulator where we can create a TokenCredential like instance using emulator available secrets (key, emulator certificate) which Cosmos backend will properly recognize for this particular case (see ifLocalEmulator).